wagtail_link_block/blocks.py

```diff
@@ -65,6 +67,12 @@ class LinkBlock(StructBlock):
         validators=[URLOrAbsolutePathValidator()],
         label=_("Custom URL"),
     )
+    anchor = CharBlock(
+        max_length=300,
+        required=False,
+        classname="anchor_link",
+        label=_("#"),
+    )
```

Comment:
Possibly need some validator on this (or protection elsewhere). All the others above limit their content with validators, this one could allow editors to add unintended code.

Reply:
What do you think could be checked ? I am not sure there is a limitation on what you can use as an anchor.

Reply:
References:
- https://html.spec.whatwg.org/multipage/browsing-the-web.html#scroll-to-the-fragment-identifier

And testing it locally, it seems you can put pretty much anything you want into it...  Even putting a second # seems to work fine - as long as the surrounding quotes end up escaped in the HTML so it doesn't break the href=""...

Reply:
Also https://www.w3.org/MarkUp/1995-archive/Elements/A.html...

@tomkins - I think they can be totally anything, as long as they don't break the end of the attribute or tag.

As long as it starts with the '#' (which `get_url` does, not the anchor...), and as long as it's escaped correctly (which the template auto-escaping should handle), I think it should be safe?

If we *do* want to limit what can go in it,we could use [RegexValidator](https://docs.djangoproject.com/en/3.2/ref/validators/#regexvalidator):

Something like:
```suggestion
    anchor = CharBlock(
        max_length=300,
        required=False,
        classname="anchor_link",
        label=_("#"),
        validators=[RegexValidator('^[^"#<>]*$')],
    )
```

thoughts?

Reply:
Any news on this ?

Reply:
@danthedeckie checked this carefully - and it seems good.

(apologies for being slow and blocking this one!)